src/app/resources/resources-add.component.ts

```diff
@@ -257,6 +262,22 @@ export class ResourcesAddComponent implements OnInit {
     this.disableDownload = false;
     this.disableDelete = false;
     this.resourceForm.updateValueAndValidity();
+
+    if (this.file.type === 'application/zip') {
+      // If the uploaded file is a zip, update attachedZipFiles to show options in openWhichFile dropdown
+
+      this.resourceForm.controls.openWhichFile.enable();
+      const fileObs: Observable<any> = this.createFileObs();
```

Comment:
This unpacks the zip file into memory, which could slow things down for larger files.  The first part of `zipObs` gets a list of the file names, so you could split that into two functions/observables and use only the file name part here.

No need to assign it to a variable, either.